src/test/checkqueue_tests.cpp

```diff
@@ -112,6 +112,7 @@ struct FrozenCleanupCheck {
         return true;
     }
     FrozenCleanupCheck() {}
```

Comment:
why is this needed?

Reply:
The default ctor is required at least here: https://github.com/bitcoin/bitcoin/blob/94a26b192f187cb50bf1ac1775b23f2b03f772b1/src/checkqueue.h#L150

Reply:
Unrelated, but is there any reason we're doing the extra move here (i.e. `push_back` instead of `emplace_back`)?

Reply:
> The default ctor is required at least here: 

Yeah, but how is the manually defined ctor different from the compiler generated?

Reply:
https://en.cppreference.com/w/cpp/language/default_constructor

> If no user-declared constructors of any kind are provided for a class type (struct, class, or union), the compiler will always declare a default constructor as an inline public member of its class.

But we have a copy ctor with this PR, so the compiler will not generate a default ctor.

Reply:
So why do you provide the copy ctor? How is it different from the compiler generated one?

Reply:
Does providing a dtor make the compiler generated ctors go away?

Reply:
> Does providing a dtor make the compiler generated ctors go away?

No, providing a destructor does not make the compiler generated constructors go away.
But the user-defined copy constructor (with this PR) makes the compiler generated default constructor go away.

My GCC fails to compile this PR without the default constructor.
It needs `FrozenCleanupCheck() {}` or `FrozenCleanupCheck() = default;`

Reply:
Idk. It passed for me after I removed all ctors

Reply:
> ... I removed all ctors

That is the point ;)

Did Clang 10 fire any warning with passed `-Wdeprecated` ?

Reply:
I don't have clang 10, and it seems to be in development. I don't think we fix compiler warnings of compilers that are in alpha stage.

Reply:
> I don't have clang 10, and it seems to be in development.

macOS 10.14 Mojave, XCode 10.0:
```
$ clang --version | grep version
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
```

Reply:
> Apple LLVM version 10.0.1 (clang-1001.0.46.4)

This version of Apple Clang is likely based off of ~ upstream Clang 6.0.1.

Reply:
> Did Clang 10 fire any warning with passed `-Wdeprecated` ?

So, I mean a compiler on macOS 10.14+ ;)

Reply:
Well, thank god I don't have macOS, which is horribly broken in every thinkable way